utils/build_stig_control.py

```diff
@@ -0,0 +1,114 @@
+
+
+SSG_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+BUILD_OUTPUT = os.path.join(SSG_ROOT, "build", "stig_control.yml")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-r", "--root", type=str, action="store", default=SSG_ROOT,
+                        help="Path to SSG root directory (defaults to %s)" % SSG_ROOT)
+    parser.add_argument("-o", "--output", type=str, action="store", default=BUILD_OUTPUT,
+                        help="File to write json output to (defaults to build/rule_dirs.json)")
+    parser.add_argument("-p", "--product", type=str, action="store", required=True,
+                        help="What product to get STIGs for")
+    parser.add_argument("-m", "--manual", type=str, action="store", required=True,
+                        help="Path to XML XCCDF manual file to use as the source of the STIGs")
+
+    return parser.parse_args()
+
+
+def collect_rule_ids_and_dirs(rules_dir):
+    for rule_dir in sorted(ssg.rules.find_rule_dirs(rules_dir)):
+        yield ssg.rules.get_rule_dir_id(rule_dir), rule_dir
+
+
+def handle_rule_yaml(product_list, product_yamls, rule_id, rule_dir, guide_dir):
+    rule_obj = {'id': rule_id, 'dir': rule_dir, 'guide': guide_dir}
+    rule_file = ssg.rules.get_rule_dir_yaml(rule_dir)
+
+    prod_type = product_list[0]
+    product_yaml = product_yamls[prod_type]
+
+    rule_yaml = ssg.build_yaml.Rule.from_yaml(rule_file, product_yaml)
+    rule_obj['references'] = rule_yaml.references
+
+    return rule_obj
+
+
+def main():
+    args = parse_args()
+    ns = {'checklist': 'http://checklists.nist.gov/xccdf/1.1'}
+    known_rules = get_implemented_stigs(args)
+    tree = ET.parse(args.manual)
+    root = tree.getroot()
+    controls = list()
+    for group in root.findall('checklist:Group', ns):
+        for stig in group.findall('checklist:Rule', ns):
+            stig_id = stig.find('checklist:version', ns).text
+            control = dict()
+            control['id'] = stig_id
+            control['levels'] = [stig.attrib['severity']]
+            control['title'] = stig.find('checklist:title', ns).text
+            if stig_id in known_rules.keys():
+                control['rules'] = known_rules.get(stig_id)
+            if 'rules' in control.keys():
+                control['zzautomated'] = 'yes'
+            else:
+                control['zzautomated'] = 'no'
+            controls.append(control)
+
+    output = dict()
+    output['controls'] = controls
+    f = open(args.output, 'w')
+    out_text = yaml.dump(output)
+    out_text = out_text.replace('zzautomated:', 'automated:')
+    f.write(out_text)
+    if not f.closed:
+        f.flush()
+        f.close()
+
+
+def get_implemented_stigs(args):
+    linux_products, other_products = ssg.products.get_all(args.root)
+    all_products = linux_products.union(other_products)
+    all_rule_dirs, product_yamls = walk_products(args.root, all_products)
```

Comment:
As an alternative, you could use `rule_dirs.json` from the other utility and then walk only rules which apply to this OS. It would be faster than walking the entire `linux_os` tree and make the behavior align better :-)